pkg/solace/resource/destination.go

```diff
@@ -171,3 +171,134 @@ func QueueNonDurableExclusiveAnonymous() *Queue {
 		durable:               false,
 	}
 }
+
+// CachedMessageSubscriptionStrategy indicates how the API should pass received cached and live messages to the application. Refer to each
+// variant for details on what behaviour they configure.
+type CachedMessageSubscriptionStrategy int
+
+const (
+	// AsAvailable provides a configuration for receiving a concurrent mix of both live and cached messages on the given TopicSubscription.
+	AsAvailable CachedMessageSubscriptionStrategy = iota
+
+	// LiveCancelsCached provides a configuration for initially passing received cached messages to the application and as soon as live
+	// messages are received, passing those instead and passing no more cached messages.
+	LiveCancelsCached
+
+	// CachedFirst provides a configuration for passing only cached messages to the application, before passing the received live messages.
+	// The live messages passed to the application thereof this configuration can be received as early as when the cache request is sent
+	// by the API, and are enqueued until the cache response is received and its associated cached messages, if available, are passed to
+	// the application.
+	CachedFirst
+
+	// CachedOnly provides a configuration for passing only cached messages and no live messages to the application.
+	CachedOnly
+)
+
+// CachedMessageSubscriptionRequest provides an interface through which cache request configurations can be constructed. These
+// configurations can then be passed to a call to a [RequestCached] interface method to request cached data. Refer to each of the below
+// factory methods for details on what configuration they provide.
+type CachedMessageSubscriptionRequest interface {
+	// GetCacheName retrieves the name of the cache.
+	GetCacheName() string
+
+	// GetName retrieves the name of the topic subscription.
+	GetName() string
+
+	GetSubscription() *TopicSubscription
+
+	GetCacheAccessTimeout() int32
+
+	GetMaxCachedMessages() int32
+
+	GetCachedMessageAge() int32
+
+	GetCachedMessageSubscriptionRequestStrategy() *CachedMessageSubscriptionStrategy
+}
+
```

Comment:
@oodigie , the name is the stringified topic name, so you don't need to duplicate the field. Please remove the `name` field and the `GetSubscription()` accessor, and instead store the configured subscription as a `*TopicSubscription` and return its string form through `GetName()` by calling `cachedMessageSubscriptionRequest.subscription.GetName()`.

Reply:
@TrentDaniel, now there is a small issue with this? Please what is returned when the passed in topic subscription argument is nil ?

Reply:
Based on our previous discussion, an empty string should be returned from `GetName()` when the TopicSubscription is nil.